apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -87,57 +81,93 @@ export function positionBlocksOnWorkspace(workspace, blockOrderMap) {
     false
   );
 
-  orderedBlocksSetupFirst.forEach(block => {
-    positionBlockWithCursor(block, cursor);
+  adjustBlockPositions(orderedBlocksSetupFirst, workspace);
+}
+
+function adjustBlockPositions(blocks, workspace) {
+  const {contentWidth = 0, viewWidth = 0} = workspace.getMetrics();
+  const {RTL} = workspace;
+
+  const padding = viewWidth ? WORKSPACE_PADDING : 0;
```

Comment:
@mikeharv I think you were using additional padding for the x-value (32 instead of 16) here. Should we increase that everywhere?

Reply:
From Mike via Slack:
>  It looks like the positioning logic is working great. One suggestion is that we should detect frame SVGs and add horizontal padding for those blocks. The vertical spacing looks good to me, but I don’t think we want the functionalSvg_s to be directly adjacent to the toolbox